### PR TITLE
Allow starting outside main thread

### PR DESCRIPTION
**Pull request by p3g4asus**
Currently telegram_menu is not runnable outside the main thread in OSes different from Windows. That's because it does not allow to customize the `stop_signals` parameter of `run_polling` function. Leaving `stop_signals` to the default value when running outside main thread under Unix systems o MacOs will raise an exception.  This PR addresses this issue. To start outside the main thread, create a thread, start it, create an new event loop inside it
`asincio.set_event_loop(asyncio.new_event_loop())`
and call start method with
`TelegramMenuSession(api_key).start(start_message_class=StartMessage, start_message_args=[params], navigation_handler_class=MyNavigationHandler, stop_signals=())`

### CODE REVIEW

**Comment by p3g4asus** [comment]
I added a commit where I fixed the `message_id` field setting of `BaseMessage` class. I also added some code to allow `update` `BaseMessage` method to be async.

File: telegram_menu/navigation.py [hunk 1]
```diff
@@ -256,6 +259,10 @@ def __init__(self, bot: Bot, chat: Chat, scheduler: BaseScheduler) -> None:
             replace_existing=True,
         )
 
+    @staticmethod
```
Comment (mevellea) [inline telegram_menu/navigation.py]: I would prefer if ´call_update´ was a member of ´BaseMessage´ instead. 
That means reworking the abstract method as well.

**Review by mevellea — CHANGES_REQUESTED** [review]
Thanks for the fix, that looks good.
I can rework the abstract class, but not this week.

File: telegram_menu/navigation.py (hunk 1 above)
Comment (p3g4asus) [reply telegram_menu/navigation.py]: OK: just wait for an additional  commit from me: I will put `call_update` as a method of `BaseMessage` and add an additional function (inlined buttons under a message with image and caption).

Comment (p3g4asus) [reply telegram_menu/navigation.py]: Please have a look to see if you like how it is implemented now.

**Comment by p3g4asus** [comment]
- Bot pickle persistence path is now customizable
- Context (optional) argument added to all callback/update/menu changing functions to allow them using bot persistence. Previous applications using the library will (hopefully) not be broken because the callback/update functions are called with context parameter only if they accept a parameter
- Added picture field to BaseMessage class to allow sending messages with picture and caption

**Comment by mevellea** [comment]
Looks better, thanks for the update. I fixed few warnings related to mypy.
One unit-test is still not passing, I can look at it but not now.

**Review by mevellea — APPROVED** [review]

**Review by mevellea — APPROVED** [review]

**Review by mevellea — APPROVED** [review]

File: telegram_menu/navigation.py [hunk 2]
```diff
@@ -298,7 +298,6 @@ async def goto_menu(
         if message is None:
             return -1  # message was not sent, abort
         menu_message.is_alive()
-        menu_message.message_id = message.message_id
```
Comment (p3g4asus) [inline telegram_menu/navigation.py]: Can I ask you why this line is wrong? Without it,  non inlined BaseMessages will get message_id always equal to -1 and so there will be no way of exiting bot software (deleting StartMessage will be impossible).  Thanks in advance for the clarification.

Comment (mevellea) [reply telegram_menu/navigation.py]: Ok I see your point, it's fixed

**Pull request merged**